app/public/src/Roll.js

```diff
@@ -0,0 +1,10 @@
+function Roll(pins) {
+  if(pins>10 || pins<0 || isNaN(pins)) {
```

Comment:
So two different perspectives here:

PERSPECTIVE 1:
The point of an initialiser is to set up a valid object. The object has been created and state is then set. I tend to have clean initialisers that expect valid input. I want input validation to happen as soon as possible to the user giving input rather than all the way down the chain to when the roll object is created. At the moment this means that if I were to change the rules I have to change the if condition of this object, which 

This also means a bunch of errors will be thrown in my program which I need to handle in order for my program not to break every single time bad input is given. Ideally, in production, i'd like to validate the input on the client side first, and then have a validation layer before calling new Roll().

PERSPECTIVE 2:

The responsibility of this object is to validate input for a roll, so it's incorrectly named. It should be some kind of RollBuilder or some such. Throwing errors is ok because it's a small specific error case that won't bubble up if handled at the correct layer.

Reply:
Understood perspective 1, I guess the problem is me building this before building the interface which means that I didn't really take into account I could have validations on the actual user interface.

Understand perspective 2, would I be able to just get away with one object then? I seem to keep having issues with naming my objects correctly (Round class in rock/paper/scissors comes to mind), any tips for that? I know it seems stupid but when I realize a pattern I get worried!